tripy/nvtripy/frontend/ops/_registry.py

```diff
@@ -32,13 +33,24 @@ def register_tensor_method(name: str):
     """
 
     # We make a special exception for "shape" since we actually do want that to be a property
-    allowed_methods = ["shape"]
+    # We also add additional methods of the tensor class that are not magic methods
+    allowed_methods = ["copy", "cast", "shape", "reshape", "transpose", "flatten", "permute", "squeeze", "unsqueeze"]
     assert name in allowed_methods or name.startswith(
         "__"
     ), f"The tensor method registry should only be used for magic methods, but was used for: {name}"
 
     def impl(func: Callable[..., Any]) -> Callable[..., Any]:
-        TENSOR_METHOD_REGISTRY[name] = func
+        if name == "shape":
+            TENSOR_METHOD_REGISTRY[name] = func
+        else:
+            # Create a method wrapper that maps 'self' to the first argument (input)
+            # This is the standard pattern for all tensor methods except 'shape' (which is a property)
+            @wraps(func)
+            def method_wrapper(self, *args, **kwargs):
+                return func(self, *args, **kwargs)
```

Comment:
It seems like this isn't doing anything? `self` should have worked fine as a variadic argument.